gammapy/datasets/map.py

```diff
@@ -1473,6 +1473,35 @@ def stat_sum(self):
         else:
             return cash_sum_cython(counts.ravel(), npred.ravel()) + prior_stat_sum
 
+    def _to_asimov_dataset(self):
+        """Create Asimov dataset from the current models.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the new dataset. Default is None.
+
+        """
+        npred = self.npred()
+        data = np.nan_to_num(npred.data, copy=True, nan=0.0, posinf=0.0, neginf=0.0)
+        npred.data = data.astype("float")
+
+        asimov_dataset = self.__class__(
+            models=self.models,
+            counts=npred,
```

Comment:
is it okay to have non integral counts?

Reply:
I mean, it is of course valid for Asimov datasets, but can we just use the same in our normal datasets?